central/notifiers/email/email.go

```diff
@@ -181,6 +183,36 @@ type message struct {
 	EmbedLogo   bool
 }
 
+// This function does not support UTF-8 strings.
+func applyRfc5322LineLengthLimit(str string) string {
+	strLen := len(str)
+
+	startPos := 0
+	numOfChunks := strLen / emailLineLength
+
+	var builder strings.Builder
+	for numOfChunks > 0 && startPos+emailLineLength < strLen {
+		builder.WriteString(str[startPos : startPos+emailLineLength])
+		builder.WriteString("\r\n")
+
+		numOfChunks--
+		startPos += emailLineLength
+	}
+	builder.WriteString(str[startPos:strLen])
+
+	return builder.String()
+}
+
+func applyRfc5322TextWordWrap(text string) string {
+	wrappedText := wordwrap.WrapString(text, emailLineLength)
+
+	// Avoid problems with already email formatted text.
+	wrappedText = strings.Replace(wrappedText, "\r\n", "\n", -1)
```

Comment:
What does this achieve? We replace \r\n with \n and then \n again with \r\n?

Reply:
For example, if the creator of text already formatted it for an email with `\r\n` and `WrapString` adds additionally `\n` to the string. Then we have a combination of `\r\n` and `\n` in the string. If we just replace `\n` with `\r\n` we will get `\r\r\n` and that breaks formatting (too many newlines).

So with this line, we are setting everything to the same line end (only `\n`) and then we apply the replacement of `\n` with `\r\n` which is correct for emails. I have added a unit test for that: `preformatted text`.